### ISSUE TITLE
Heimdal won't open files that are o+r and g+r

### ISSUE THREAD

**Issue by antekone** [msys2/MSYS2-packages#1480]
When using `kinit`, Heimdal displays this error:

    kinit: krb5_init_creds_store: Refuses to open group/other readable files FILE:/tmp/krb5cc_jXmWbn

The problem is that I can't perform `chmod go-r` on this file, because it seems that the `go-r` permission setting isn't supported by msys2? Even when I'll do `chmod 600 <file>`, the permissions are still `go+r`.

Is there any workaround I can do to be able to run `kinit` again?

I can't compile my own Heimdal, because compilation fails on problems with compiling one of its dependencies, `libedit` (missing `termios.h`).

So far I've simply patched the code segment `/bin/msys-krb5-26.dll` to manually remove the offending jump, and it works, but obviously this will become a problem after next update.

**Comment by mingwandroid** [comment]
You need to fix it in the upstream source code, submit a patch to them, then add PR here with the patch.

**Comment by antekone** [comment]
I've managed to compile Heimdal under MSYS. I've tried to do it from MinGW earlier, hence the errors during compilation.

I've checked the issue on the upstream side. They do support Windows, as the permission checking & failing code is placed between the `#ifndef _WIN32` blocks. So when compiling i.e. on Visual Studio, they work fine. They just don't support MSYS, since MSYS doesn't define `_WIN32`. So the organizational question is who should support this, MSYS or Heimdal (I honestly don't know, this depends on your point of view only I guess)?

Another thing is that it seems this permission check can be disabled by configuration option. After adding `fcache_strict_checking = false` to `/etc/krb5/krb5.conf`, the check is skipped on MSYS as well, so this could be one workaround. If there is no such option defined, its default value is `true`. If so, another question is, should this option be set to `false` on MSYS' distributed `krb5.conf` by default? Because the package is currently not working at all by default. 

Third option is to add a patch for Heimdal MSYS2 package that changes the default setting for `fcache_strict_checking`. Currently when the option is not specified, this means `true`. After changing it to `false`, the default installation of Heimdal will work, no fixes in upstream will be needed and no configuration will need to be changed.

**Comment by mingwandroid** [comment]
Why are you compiling using the msys subsystem's compilers? This generates, rougly speaking, Cygwin-y software which is slow and always GPL2. You should use our mingw-w64 instead.

**Comment by antekone** [comment]
I'm compiling it with a MSYS2 compiler because that's what I think was used to compile the original package, `heimdal`, that is the owner of `/usr/bin/kinit.exe` and `/usr/bin/klist.exe` files.

I've cloned `MSYS2-packages` repository, entered `Heimdall` directory, and invoked `makepkg` in order to compile the project. I've done this from the MSYS2 shell. When trying to compile it with mingw-w64, the compilation was failing on one of the dependencies, `libedit`. It did compile on MSYS2 without issues (after applying your set of patches).

It looked natural for me to do it this way, because when I've inspected the original `/usr/bin/kinit.exe`, which is installed by issuing `pacman -S heimdal`, it seems to be compiled with MSYS2 gcc instead of mingw-w64 gcc:

    $ ldd /usr/bin/kinit.exe
            ntdll.dll => /c/WINDOWS/SYSTEM32/ntdll.dll (0x7ff960d10000)
            KERNEL32.DLL => /c/WINDOWS/System32/KERNEL32.DLL (0x7ff95e480000)
            KERNELBASE.dll => /c/WINDOWS/System32/KERNELBASE.dll (0x7ff95da10000)
            ...
            msys-2.0.dll => /usr/bin/msys-2.0.dll (0x180040000)
            ...
            msys-sqlite3-0.dll => /usr/bin/msys-sqlite3-0.dll (0x5798a0000)
            msys-gcc_s-seh-1.dll => /usr/bin/msys-gcc_s-seh-1.dll (0x5e8160000)
            msys-z.dll => /usr/bin/msys-z.dll (0x522fe0000)

So in order to find a way to fix the issue described in my original first post, I had to find a way to create the `/usr/bin/kinit.exe`, which I think is to simply use an MSYS2 compiler.
